### DISCUSSION TITLE
Date Calculations in views

### DISCUSSION THREAD

**Discussion by TheBig-O** [mabeyj/trilium-collection-views#56] [General]
I was looking for a way to calculate date differences within the table view. For example, if the due date was 1 November and the date completed was 1 December, is there a way to show the 30-day difference?   
Another example would be showing how long ago a task was completed. This would require a calculation between today and the date completed?
I'm happy to try to figure out how to do this but don't really know where to start. I looked for the calculations you already do for "Time Spent" and can't sort out where that calculation is coming from.
Any help you can provide would be awesome!
Thanks for building this. I use it every day!

**Comment by mabeyj** [comment]
Hi, this plugin doesn't support making calculations like this.

I would guess the way this could be done would be to write a script that calculates `today - dateCompleted` on each of your notes and stores the result in a separate `timeSinceCompleted` attribute. Then, you would display `timeSinceCompleted` in the table view. Since this would change every day, the script would probably need to be run daily which could be accomplished using a daily global event: https://github.com/zadam/trilium/wiki/Events

The "Time spent" doesn't come from this plugin. It's probably from the To-do plugin: https://github.com/zadam/trilium/discussions/2946

I don't use it myself, but looking into the zip file, it looks like `To Do/Scripts/attributeChanged/taskComplete/taskStopTracking/taskUpdateTimeSpent.js` is the script responsible for that calculation.

**Reply by TheBig-O** [reply]
Thanks for the reply. I should have realized that it was coming from the ToDo plugin. I use them all interchangeably and got confused.
I'll take a look at the `taskUpdateTimeSpent.js` and see if I can sort out a way to add that capability. Unfortunately, the ToDo plugin is abandonware so we have to decipher it on our own.
